jobfunnel/glassdoor_static.py

```diff
@@ -0,0 +1,271 @@
+                )
+
+                fts.append(  # append thread job future to futures list
+                    threads.submit(
+                        self.search_page_for_job_soups,
+                        data,
+                        page,
+                        page_url,
+                        job_soup_list,
+                    )
+                )
+        wait(fts)  # wait for all scrape jobs to finish
+
+        # make a dict of job postings from the listing briefs
+        for s in job_soup_list:
+            # init dict to store scraped data
+            job = dict([(k, "") for k in MASTERLIST_HEADER])
+
+            # scrape the post data
+            job["status"] = "new"
+            try:
+                # jobs should at minimum have a title, company and location
+                job["title"] = (
+                    s.find("div", attrs={"class", "jobContainer"})
+                    .find(
+                        "a",
+                        attrs={"class", "jobLink jobInfoItem jobTitle"},
+                        recursive=False,
+                    )
+                    .text.strip()
+                )
+                job["company"] = s.find(
+                    "div", attrs={"class", "jobInfoItem jobEmpolyerName"}
+                ).text.strip()
+                job["location"] = s.get("data-job-loc")
+            except AttributeError:
+                continue
+
+            # set blurb to none for now
+            job["blurb"] = ""
+
+            try:
+                labels = s.find_all("div", attrs={"class", "jobLabel"})
+                job["tags"] = "\n".join(
+                    [l.text.strip() for l in labels if l.text.strip() != "New"]
+                )
+            except AttributeError:
+                job["tags"] = ""
+
+            try:
+                job["date"] = (
+                    s.find("div", attrs={"class", "jobLabels"})
+                    .find("span", attrs={"class", "jobLabel nowrap"})
+                    .text.strip()
+                )
+            except AttributeError:
+                job["date"] = ""
+
+            try:
+                part_url = (
+                    s.find("div", attrs={"class", "logoWrap"}).find(
+                        "a").get("href")
+                )
+                job["id"] = s.get("data-id")
+                job["link"] = (
+                    f"https://www.glassdoor."
+                    f"{self.search_terms['region']['domain']}"
+                    f"{part_url}"
+                )
+
+            except (AttributeError, IndexError):
+                job["id"] = ""
+                job["link"] = ""
+
+            job["query"] = self.query
+            job["provider"] = self.provider
+
+            # key by id
+            self.scrape_data[str(job["id"])] = job
+
+        # Do not change the order of the next three statements if you want date_filter to work
+
+        # stores references to jobs in list to be used in blurb retrieval
```

Comment:
I think this logic from 247 on should be in GlassDoorBase since it is duplicated.